### ISSUE TITLE
Inconsistent planemo test output (silent failures)

### ISSUE THREAD

**Issue by peterjc** [galaxyproject/planemo#291]
I've been trying ``planemo test`` with TravisCI on a sub-set of https://github.com/peterjc/pico_galaxy in a new experimental repository https://github.com/peterjc/galaxy_mira

See: https://travis-ci.org/peterjc/galaxy_mira/builds/79507686 (overview of test run 18)

Sub-job 18.1, https://travis-ci.org/peterjc/galaxy_mira/jobs/79507687 and sub-job 18.2 https://travis-ci.org/peterjc/galaxy_mira/jobs/79507689 look good (Python 2.6 using galaxy dev and master):

```
All 17 test(s) executed passed.
mira_4_0_bait[0]: passed
mira_4_0_bait[1]: passed
mira_4_0_bait[2]: passed
mira_4_0_convert[0]: passed
mira_4_0_convert[1]: passed
mira_4_0_de_novo[0]: passed
mira_4_0_de_novo[1]: passed
mira_4_0_mapping[0]: passed
mira_4_0_mapping[1]: passed
mira_4_9_bait[0]: passed
mira_4_9_bait[1]: passed
mira_4_9_bait[2]: passed
mira_4_9_bait[3]: passed
mira_4_9_bait[4]: passed
mira_4_9_bait[5]: passed
mira_4_9_bait[6]: passed
mira_assembler[0]: passed
The command "planemo test ${TRAVIS_BUILD_DIR}/tools/" exited with 0.
```

Sub job 18.3 https://travis-ci.org/peterjc/galaxy_mira/jobs/79507691 is confused (Python 2.7 and galaxy dev), note this says *mira_4_9_bait[5]* failed.

```
All 17 test(s) executed passed.
mira_4_0_bait[0]: passed
mira_4_0_bait[1]: passed
mira_4_0_bait[2]: passed
mira_4_0_convert[0]: passed
mira_4_0_convert[1]: passed
mira_4_0_de_novo[0]: passed
mira_4_0_de_novo[1]: passed
mira_4_0_mapping[0]: passed
mira_4_0_mapping[1]: passed
mira_4_9_bait[0]: passed
mira_4_9_bait[1]: passed
mira_4_9_bait[2]: passed
mira_4_9_bait[3]: passed
mira_4_9_bait[4]: passed
mira_4_9_bait[5]: failed
mira_4_9_bait[6]: passed
mira_assembler[0]: passed
The command "planemo test ${TRAVIS_BUILD_DIR}/tools/" exited with 0.
```

Sub job 18.4 https://travis-ci.org/peterjc/galaxy_mira/jobs/79507693 is also confused (Python 2.7 and Galaxy stable), note this says *mira_4_0_mapping[1]* failed:

```
All 17 test(s) executed passed.
mira_4_0_bait[0]: passed
mira_4_0_bait[1]: passed
mira_4_0_bait[2]: passed
mira_4_0_convert[0]: passed
mira_4_0_convert[1]: passed
mira_4_0_de_novo[0]: passed
mira_4_0_de_novo[1]: passed
mira_4_0_mapping[0]: passed
mira_4_0_mapping[1]: failed
mira_4_9_bait[0]: passed
mira_4_9_bait[1]: passed
mira_4_9_bait[2]: passed
mira_4_9_bait[3]: passed
mira_4_9_bait[4]: passed
mira_4_9_bait[5]: passed
mira_4_9_bait[6]: passed
mira_assembler[0]: passed
The command "planemo test ${TRAVIS_BUILD_DIR}/tools/" exited with 0.
```

Notice some of these test runs had a failure (not consistently), yet planemo still says all 17 tests passed, and returns 0 overall.

**Comment by peterjc** [comment]
This might be a duplicate of #197 ?

**Comment by peterjc** [comment]
P.S. I really like the planemo's red/green highlighting of key messages in the terminal output works on the TravisCI rendering of the captured output - it makes these silent failures very visible.

**Comment by mvdbeek** [comment]
Could this be related to issue #280 and PR #281 ?

**Comment by peterjc** [comment]
@mvdbeek I suppose I could have some bad XML hidden by #280 or #281, but I think this is a different problem.

**Comment by mvdbeek** [comment]
@peterjc you're right, the referenced issue is specifically about linting.

**Comment by erasche** [comment]
@peterjc would you mind re-testing now that @bgruening found something and it was patched? xref https://github.com/galaxyproject/planemo/issues/363#issuecomment-154820426

**Comment by peterjc** [comment]
I can ask TravisCI to rerun the ambiguous ones, in the hope for a clear answer if that's what you meant. It will lose the original output but I copy-and-pasted the key points into this issue report anyway.

**Comment by erasche** [comment]
@peterjc hmm instead of losing said output, I can just run those by hand when I find time. :)

**Comment by jmchilton** [comment]
I believe this will be fixed with #429 - I'll push an update to PyPI later today. Please reopen if this doesn't address the problem.
